pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -183,10 +183,38 @@ func toSelectableFields(claim *resource.ResourceClaim) fields.Set {
 
 // dropDisabledFields removes fields which are covered by a feature gate.
 func dropDisabledFields(newClaim, oldClaim *resource.ResourceClaim) {
+	dropDisabledDRAPrioritizedListFields(newClaim, oldClaim)
 	dropDisabledDRAAdminAccessFields(newClaim, oldClaim)
 	dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim)
 }
 
+func dropDisabledDRAPrioritizedListFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAPrioritizedList) {
+		return
+	}
+	if draPrioritizedListFeatureInUse(oldClaim) {
+		return
+	}
+
+	for i := range newClaim.Spec.Devices.Requests {
+		newClaim.Spec.Devices.Requests[i].FirstAvailableOf = nil
+	}
+}
+
+func draPrioritizedListFeatureInUse(claim *resource.ResourceClaim) bool {
+	if claim == nil {
+		return false
+	}
+
+	for _, request := range claim.Spec.Devices.Requests {
+		if len(request.FirstAvailableOf) == 0 {
```

Comment:
Is this passing the tests? Then the tests must be incomplete, because the logic is inverted here.

```suggestion
		if len(request.FirstAvailableOf) > 0 {
```

Reply:
Good catch. I've fixed the issue and added another test case that covers this.